### PR TITLE
feat: ga4 hybrid mode support

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change
- ga4 hybrid mode support added
- hybrid mode : useNativeSDK - true and useNativeSDKToSend - false

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ✅] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [✅ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/731)

### CODE REVIEW

**Review by MoumitaM — APPROVED** [review]

File: integrations/GA4/browser.js [hunk 1]
```diff
@@ -15,13 +15,17 @@ import { NAME } from "./constants";
...
+      logger.setLogLevel(analytics.logLevel);
+    }
     this.measurementId = config.measurementId;
     this.analytics = analytics;
     this.sendUserId = config.sendUserId || false;
     this.blockPageView = config.blockPageViewEvent || false;
     this.extendPageViewParams = config.extendPageViewParams || false;
     this.extendGroupPayload = config.extendGroupPayload || false;
     this.debugMode = config.debugMode || false;
+    this.isHybridModeEnabled = !config.useNativeSDKToSend;
```
Comment (arnab-p) [inline integrations/GA4/browser.js:28]: What happens if `useNativeSDKToSend` is `null` or `undefined`?

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: integrations/GA4/browser.js (hunk 1 above)
Comment (mihir-4116) [reply integrations/GA4/browser.js]: updated

File: integrations/GA4/browser.js [hunk 2]
```diff
@@ -12,16 +12,23 @@ import {
...
     this.measurementId = config.measurementId;
     this.analytics = analytics;
     this.sendUserId = config.sendUserId || false;
     this.blockPageView = config.blockPageViewEvent || false;
     this.extendPageViewParams = config.extendPageViewParams || false;
     this.extendGroupPayload = config.extendGroupPayload || false;
     this.debugMode = config.debugMode || false;
+    this.isHybridModeEnabled = isDefinedAndNotNull(config.useNativeSDKToSend)
+      ? !config.useNativeSDKToSend
+      : false;
```
Comment (arnab-p) [inline integrations/GA4/browser.js:31]:
```suggestion
    this.isHybridModeEnabled = (config.useNativeSDKToSend === false) || false;
```

Can it be like this?

Comment (mihir-4116) [reply integrations/GA4/browser.js]: yes, updated

**Review by arnab-p — DISMISSED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by ItsSudip — COMMENTED** [review]
Can we update the logger here? You can take reference from [this](https://github.com/rudderlabs/rudder-sdk-js/pull/657).

**Comment by mihir-4116** [comment]
> Can we update the logger here? You can take reference from [this](https://github.com/rudderlabs/rudder-sdk-js/pull/657).

updated

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**